database/src/main/java/database/dao/EntityDaoBase.java

```diff
@@ -0,0 +1,115 @@
+package database.dao;
+
+import database.entity.BaseEntity;
+import database.exception.EmptyValueException;
+import database.exception.IdProvidedManuallyException;
+import database.exception.IncorrectPropertyNameException;
+import database.exception.NullOrEmptyListException;
+import database.exception.NullPropertyNameOrValueException;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides a base implementation for data access operations on entities, offering common methods
+ * for database operations.
+ *
+ * @author <a href='mailto:shashinadya@gmail.com'>Nadya Shashina</a>
+ */
+public abstract class EntityDaoBase implements EntityDao {
+    public static final String ENTITY_IS_NOT_FOUND = "Entity with provided Id does not exist";
+    public static final String ID_PROVIDED_MANUALLY = "User cannot provide id manually. Ids are filled automatically.";
+    public static final String INVALID_PARAMETER_VALUE = "Invalid parameter value. " +
+            "Limit value should be in(0..{MAX_LIMIT_VALUE}), offset value should be >= 0";
+    public static final String ENTITIES_LIST_NULL_OR_EMPTY = "Entities list cannot be null or empty";
+    public static final String IDS_LIST_NULL_OR_EMPTY = "IDs list cannot be null or empty";
+    public static final String FILTER_CANNOT_BE_NULL_MESSAGE = "Property name and value cannot be null";
+    public static final String FILTER_CANNOT_BE_EMPTY_MESSAGE = "Value cannot be empty";
+    public static final String INCORRECT_FILTER_NAME_MESSAGE = "Incorrect filter name";
+
```

Comment:
We can't use it in all of those classes, because of the following reasons:

1. `DatabaseServiceRestController` and `EntityDaoRestControllerTest` classes are not related to `EntityDaoBase`, this DAO shouldn't even know about REST layer! Please don't mix logic layer and view layer!
2. `MySqlEntityDao` is an isolated class managing connections. I think we also don't need to use constants from DAO class here.